app/bundles/LeadBundle/Form/Type/ActionRemoveDoNotContact.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * @copyright   2014 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Form\Type;
+
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * Class FormSubmitActionRemoveDoNotContact.
+ */
+class ActionRemoveDoNotContact extends AbstractType
+{
+    private $factory;
+
+    /**
+     * @param MauticFactory $factory
+     */
+    public function __construct(MauticFactory $factory)
+    {
+        $this->factory = $factory;
```

Comment:
We are trying to remove MauticFactory from Mautic and replace it with DI to make the code easier to test. Could you remove this dependency since it's not needed in the class? I know you just copied similar form type, but let's not add factory dependencies in the new code.